pkg/ignition/nmstate.go

```diff
@@ -22,7 +23,7 @@ func nmstateOutputToFiles(generatedConfig []byte) ([]ignition_config_types_32.Fi
 	}
 	for _, v := range networkManagerConfig.NetworkManager {
 		name := v[0]
-		source := "data:," + v[1]
+		source := "data:," + url.QueryEscape(v[1])
```

Comment:
I think `QueryEscape` actually [differs](https://github.com/vincent-petithory/dataurl/blob/master/rfc2396.go#L11) slightly from what we want, in that it encodes spaces as `+` but we want them encoded as `%20`.

Ignition uses the library https://github.com/vincent-petithory/dataurl for [validation](https://github.com/coreos/ignition/blob/main/config/v3_2/types/url.go#L20); it would probably be best if we used the same library for encoding.

Reply:
Confusingly in Python the default quote function argument to `urlencode()` is `quote_plus` (same as `QueryEscape` in Go), but the `urlencode` filter in Jinja2 (as used in Dmitry's original) [does _not_](https://stackoverflow.com/questions/33450404/quote-plus-url-encode-filter-in-jinja2).